lib/dsc-lib/src/util.rs

```diff
@@ -232,6 +233,29 @@ pub fn resource_id(type_name: &str, name: &str) -> String {
     result
 }
 
+pub fn canonicalize_which(executable: &str, cwd: Option<&str>) -> Result<String, DscError> {
+    // Use PathBuf to handle path separators robustly
+    let mut executable_path = PathBuf::from(executable);
+    if cfg!(target_os = "windows") && executable_path.extension().is_none() {
+        executable_path.set_extension("exe");
+    }
+    let mut executable = executable_path.to_string_lossy().to_string();
+    if which(&executable).is_err() && !Path::new(&executable).is_absolute() {
+        if let Some(cwd) = cwd {
+            let cwd_path = Path::new(cwd);
+            match canonicalize(cwd_path.join(&executable)) {
```

Comment:
Wait, why aren't we just using `which_in`? https://docs.rs/which/latest/which/fn.which_in.html

> Find binary_name in the path list paths, using cwd to resolve relative paths.

That's exactly what this PR sets out to do, no? If so, let's use it if there's no reason not to!

Reply:
`which_in()` needs to separate out the exe, rel path, and cwd, so I don't think it saves anything but probably adds more code than using `canonical()` to resolve relative path.